tests/functional/fs/test_fs_statistics.py

```diff
@@ -0,0 +1,171 @@
+#
+# Copyright(c) 2019 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause-Clear
+#
+
+import time
+import pytest
+
+from core.test_run import TestRun
+from test_tools.disk_utils import Filesystem
+from test_utils.os_utils import sync
+from test_tools.fs_utils import create_directory
+from test_tools.fio.fio import Fio
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from test_tools.fio.fio_result import FioResult
+from test_utils.size import Size, Unit
+from api.iotrace_stats_parser import (
+    parse_fs_stats,
+    DirectoryTraceStatistics,
+    FileTraceStatistics,
+    ExtensionTraceStatistics,
+)
+
+
+@pytest.fixture(params=Filesystem)
+def fs(request):
+    file_system = request.param
+    mountpoint = "/mnt"
+    disk = TestRun.dut.disks[0]
+
+    with TestRun.step(f"Create file system {file_system}"):
+        disk.create_filesystem(file_system)
+
+    with TestRun.step("Mount device"):
+        disk.mount(mountpoint)
+
+    yield disk, mountpoint
+
+    with TestRun.step("Unmount device"):
+        disk.unmount()
+
+
+def test_fs_operations(fs):
+    iotrace = TestRun.plugins["iotrace"]
+
+    (disk, filesystem) = fs
+
+    dir_Ax_size = Size(3, Unit.MiB)
+    dir_Ax_WiF = 2
+    dir_Ax_written = dir_Ax_size * dir_Ax_WiF
+
+    dir_By_size = Size(16, Unit.KiB)
+    dir_By_WiF = 2
+    dir_By_written = dir_By_size * dir_By_WiF
+
+    Ay_size = Size(2, Unit.MiB)
+    Ay_WiF = 5
+    Ay_written = Ay_size * Ay_WiF
+
+    Bx_size = Size(5, Unit.MiB)
+    Bx_WiF = 1
+    Bx_written = Bx_size * Bx_WiF
+
+    A_prefix_workset = dir_Ax_size + Ay_size
+    B_prefix_workset = dir_By_size + Bx_size
+
+    A_prefix_WiF = (dir_Ax_written + Ay_written) / A_prefix_workset
+    B_prefix_WiF = (dir_By_written + Bx_written) / B_prefix_workset
+
+    x_extension_workset = dir_Ax_size + Bx_size
+    y_extension_workset = dir_By_size + Ay_size
+
+    x_extension_WiF = (dir_Ax_written + Bx_written) / x_extension_workset
+    y_extension_WiF = (Ay_written + dir_By_written) / y_extension_workset
+
+    test_dir_workset = dir_Ax_size + dir_By_size
+    root_dir_workset = Ay_size + Bx_size
+
+    test_dir_WiF = (dir_Ax_written + dir_By_written) / test_dir_workset
+    root_dir_WiF = (Ay_written + Bx_written) / root_dir_workset
+
+    with TestRun.step("Prepare files and directories for test"):
+        test_dir = f"{filesystem}/test_dir"
+
+        create_directory(test_dir)
+
+        fio_cfg = (
+            Fio()
+            .create_command()
+            .io_engine(IoEngine.libaio)
+            .sync(True)
+            .read_write(ReadWrite.write)
+            .group_reporting(False)
+            .create_only(True)
+        )
+
+        (
+            fio_cfg.add_job("test_dir/A.x")
+            .target(f"{test_dir}/A.x")
+            .file_size(dir_Ax_size)
+            .block_size(Size(4, Unit.KiB))
```

Comment:
why each job has bigger block_size?

Reply:
Just to mess with the tracer :) I thought it might be worthwile to look for bugs connected with blocksize